python/stretch_factory/firmware_updater.py

```diff
@@ -621,22 +622,64 @@ def set_target_from_install_version(self):
                     click.secho('---------- %s [%s]-----------' % (
                     device_name.upper(), str(self.fw_installed.get_version(device_name))), fg="blue", bold=True)
                     default_id = 0
-                    for i in range(len(vs)):
+
+                    ## Checks to hw id to ensure that user can not downgrade fw to far
+                    for f_limit in range(len(vs)):  
+                        if self.fw_installed.get_hw_id(device_name) >= 3:
+                            fw_limit = '0.7.0p5'
+                            fw_version = str(vs[f_limit])
+                            fw_version = fw_version[fw_version.index('v') + 1:]
+                            if fw_version == fw_limit:
+                                click.secho(f"Can only install firmware versions from {fw_limit} onwards", fg="yellow")
+                                break
+                        else:
+                            fw_limit = '0.3.1p2'
+                            fw_version = str(vs[f_limit])
+                            fw_version = fw_version[fw_version.index('v') + 1:]
+                            if fw_version == fw_limit:
+                                click.secho(f"Can only install firmware versions from {fw_limit} onwards", fg="yellow")
```

Comment:
It is not necessary to print this out. It's implied that the tool would only show the versions the PCB is allowed to install.